.github/workflows/ci-tests.yml

```
@@ -35,6 +35,13 @@ jobs:
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python_version }}
+        cache: pip
+
+    - uses: actions/cache@v3
+      if: startsWith(runner.os, 'Linux')
+      with:
+        path: ~/.cache/pre-commit
+        key: ${{ runner.os }}|pre-commit
```

let's at least use the `.pre-commit-config.yaml`'s content, otherwise it'll (un)cache the current versions forever.
  ```suggestion
        key: pre-commit-${{ hashFiles('.pre-commit-config.yaml') }}
```